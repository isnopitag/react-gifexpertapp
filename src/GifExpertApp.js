import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory.js";
import { GifGrid } from "./components/GifGrid.js";

export const GifExpertApp = ( {defaultCategories = []} ) => {
  // const [categories, setCategories] = useState(["evangelion"]);
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />
      <AddCategory setCategories={setCategories} />
      <ol>
        {categories.map((category) => (
          <GifGrid 
            category={category}
            key={category}
            />
        ))}
      </ol>
    </>
  );
};


