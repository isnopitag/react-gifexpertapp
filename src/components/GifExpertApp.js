import React, { useState } from "react";
import AddCategory from "./AddCategory.js";
import { GifGrid } from "./GifGrid.js";

const GifExpertApp = () => {
  const [categories, setCategories] = useState(["evangelion"]);

  return (
    <>
      <h2>GifExpertApp</h2>
      <hr />
      <AddCategory setCategories={setCategories} />
      <ol>
        {categories.map((category) => (
          <GifGrid 
            category={category}
            key={category}
            />
        ))}
      </ol>
    </>
  );
};

export default GifExpertApp;
