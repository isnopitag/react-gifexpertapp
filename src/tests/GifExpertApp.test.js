import React from "react";
import { GifExpertApp } from "../GifExpertApp";
import { shallow } from "enzyme";

describe("Pruebas en GifExpertApp", () => {
  
    test("debe de mostrar <GifExpertApp/>", () => {
      const wrapper = shallow(<GifExpertApp />);
  
      expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar una lista de categorias', () => {

      const categories = ['Evangelion'];
      const wrapper = shallow(<GifExpertApp  defaultCategories={categories}/>);
      expect(wrapper).toMatchSnapshot();
      expect( wrapper.find('GifGrid').length ).toBe( categories.length );


    })
    

    
})