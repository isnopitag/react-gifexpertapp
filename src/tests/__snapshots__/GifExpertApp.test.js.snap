// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="Evangelion"
      key="Evangelion"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en GifExpertApp debe de mostrar <GifExpertApp/> 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</Fragment>
`;
